From 90548f3e37e66931f1042c2b24d9c59d5d83b38e Mon Sep 17 00:00:00 2001
From: David Redondo <kde@david-redondo.de>
Date: Tue, 29 Sep 2020 14:57:37 +0200
Subject: [PATCH] Fix crash with text face

A binding involving the two arrays won't update when we change just the
contents of the arrays so we crashed later in QQmlData::wasDeleted when
trying to access the objects which were inside but deleted.
---
 .../textonly/contents/ui/GroupedText.qml      | 20 +++++++++++++++----
 1 file changed, 16 insertions(+), 4 deletions(-)

diff --git a/faces/facepackages/textonly/contents/ui/GroupedText.qml b/faces/facepackages/textonly/contents/ui/GroupedText.qml
index 94f8a3c..b2cd09d 100644
--- a/faces/facepackages/textonly/contents/ui/GroupedText.qml
+++ b/faces/facepackages/textonly/contents/ui/GroupedText.qml
@@ -104,8 +104,14 @@ ColumnLayout {
 
         Sensors.Sensor { sensorId: modelData }
 
-        onObjectAdded: root.highPrioritySensors.push(object)
-        onObjectRemoved: root.highPrioritySensors.splice(root.highPrioritySensors.indexOf(object), 1)
+        onObjectAdded: {
+            root.highPrioritySensors.push(object)
+            root.highPrioritySensors = root.highPrioritySensors
+        }
+        onObjectRemoved: {
+            root.highPrioritySensors.splice(root.highPrioritySensors.indexOf(object), 1)
+            root.highPrioritySensors = root.highPrioritySensors
+        }
     }
 
     Instantiator {
@@ -113,7 +119,13 @@ ColumnLayout {
 
         Sensors.Sensor { sensorId: modelData }
 
-        onObjectAdded: root.lowPrioritySensors.push(object)
-        onObjectRemoved: root.lowPrioritySensors.splice(root.lowPrioritySensors.indexOf(object), 1)
+        onObjectAdded: {
+            root.lowPrioritySensors.push(object)
+            root.lowPrioritySensors = root.lowPrioritySensors
+        }
+        onObjectRemoved: {
+            root.lowPrioritySensors.splice(root.lowPrioritySensors.indexOf(object), 1)
+            root.lowPrioritySensors = root.lowPrioritySensors
+        }
     }
 }
-- 
GitLab

